src/aeso_ast_infer_types.erl

```diff
@@ -846,7 +854,25 @@ infer1(Env, [{Contract, Ann, ConName, Code} | Rest], Acc, Options)
         contract_interface -> ok
     end,
     {Env1, Code1} = infer_contract_top(push_scope(contract, ConName, Env), What, Code, Options),
-    Contract1 = {Contract, Ann, ConName, Code1},
+    Contract1 = {Contract, Ann, ConName, Impls, Code1},
+    AllInterfaces = [{name(IName), I} || I = {contract_interface, _, IName, _, _} <- Acc],
+    ImplsNames = lists:map(fun name/1, Impls),
+    create_type_errors(),
+    lists:foreach(fun(Impl) -> case proplists:get_value(name(Impl), AllInterfaces) of
+                                   undefined -> type_error({referencing_undefined_interface, Impl});
+                                   _         -> ok
+                               end
+                  end, Impls),
+    destroy_and_report_type_errors(Env),
```

Comment:
Removing that was causing a failed test, but I've just found out that the failing test was failing because of another mistake that I've made.

Fixed here: 4790566f427a3f9ec01b41a61217a7fe319d2525